### PR TITLE
Add min_pix option in WcsGeom.cutout and set it to 3 for IRFs 

### PR DESCRIPTION
**Pull request by QRemy**
Add min_pix option in WcsGeom.cutout and set it to 3 for IRFs in order to ensure that IRF interpolation is done correctly if the binning of the IRFs is larger than the width of the analysis region.

### CODE REVIEW

**Comment by bkhelifi** [comment]
Thanks @QRemy... What happens where there is not enough bin for the interpolation?

**Comment by QRemy** [comment]
> What happens where there is not enough bin for the interpolation?
It uses the single bin that is left in the psf cutout. So if we reduce the region size the psf used at a given position can be different to the one used before cutout.

**Review by bkhelifi — APPROVED** [review]
Thanks @QRemy! Looks good for me

File: gammapy/maps/wcs/ndmap.py [hunk 1]
```diff
@@ -992,14 +992,21 @@ def cutout(self, position, width, mode="trim", odd_npix=False):
         odd_npix : bool, optional
             Force width to odd number of pixels.
             Default is False.
+        min_npix : bool, optional
```
Comment (registerrier) [inline gammapy/maps/wcs/ndmap.py:995]:
```suggestion
        min_npix : int, optional
```

File: gammapy/maps/wcs/ndmap.py [hunk 2]
```diff
@@ -992,14 +992,21 @@ def cutout(self, position, width, mode="trim", odd_npix=False):
...
+            Force width to a minimmum number of pixels.
```
Comment (registerrier) [inline gammapy/maps/wcs/ndmap.py:996]:
```suggestion
            Force width to a minimum number of pixels.
```

Comment (registerrier) [reply gammapy/maps/wcs/ndmap.py:996]: Maybe specify that the minimum number applies in both axes.

File: gammapy/irf/core.py [hunk 3]
```diff
@@ -958,9 +958,10 @@ def cutout(self, position, width, mode="trim"):
         cutout : `IRFMap`
             Cutout IRF map.
         """
-        irf_map = self._irf_map.cutout(position, width, mode)
+
+        irf_map = self._irf_map.cutout(position, width, mode, min_npix=3)
```
Comment (registerrier) [inline gammapy/irf/core.py:962]: I think the option should be explicitly available on the `cutout` arguments. A minima it should be explained that the minimum cutout is 3 pixels wide.

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy . I have left a few inline comments.
I think I understand the use case. I just wonder whether it should be configurable as an argument of the `IRFMap.cutout` method. The docstring could briefly explain why 3 is a good default. No?

Does it need to be backported?

File: gammapy/irf/core.py (hunk 3 above)
Comment (QRemy) [reply gammapy/irf/core.py]: done

**Pull request merged**